be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -105,6 +105,336 @@ private:
     }
 };
 
+template <PrimitiveType PT>
+class ArrayDifference {
+public:
+    using CppType = RunTimeCppType<PT>;
+
+    static ColumnPtr process(FunctionContext* ctx, const Columns& columns) {
+        if constexpr (pt_is_float<PT> || pt_is_integer<PT> || pt_is_decimalv2<PT> || pt_is_boolean<PT>) {
+            return _array_difference(columns);
+        } else {
+            assert(false);
+        }
+    }
+
+private:
+    static ColumnPtr _array_difference(const Columns& columns) {
+        DCHECK_EQ(columns.size(), 1);
+
+        size_t chunk_size = columns[0]->size();
+        ColumnPtr src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
+        ColumnPtr dest_column_data = nullptr;
+        ColumnPtr dest_column = nullptr;
+
+        if constexpr (pt_is_float<PT>) {
+            dest_column_data = NullableColumn::create(DoubleColumn::create(), NullColumn::create());
+        } else if constexpr (pt_is_integer<PT> || pt_is_boolean<PT>) {
```

Comment:
the result of array_differenct([false, true, false]) ?

Reply:
[0, 1, -1]